app/controllers/declaration/IsLicenseRequiredController.scala

```diff
@@ -0,0 +1,96 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.declaration
+
+import controllers.actions.{AuthAction, JourneyAction}
+import controllers.declaration.routes._
+import controllers.navigation.Navigator
+import forms.common.YesNoAnswer
+import forms.common.YesNoAnswer.YesNoAnswers
+import forms.common.YesNoAnswer.YesNoAnswers.{no, yes}
+import forms.declaration.declarationHolder.AuthorizationTypeCodes
+import models.DeclarationType._
+import models.requests.JourneyRequest
+import models.{ExportsDeclaration, Mode}
+import play.api.data.Form
+import play.api.i18n.I18nSupport
+import play.api.mvc._
+import services.cache.ExportsCacheService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import views.html.declaration.is_license_required
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class IsLicenseRequiredController @Inject()(
+  authenticate: AuthAction,
+  journeyType: JourneyAction,
+  override val exportsCacheService: ExportsCacheService,
+  navigator: Navigator,
+  mcc: MessagesControllerComponents,
+  is_license_required: is_license_required
+)(implicit ec: ExecutionContext)
+    extends FrontendController(mcc) with I18nSupport with ModelCacheable with SubmissionErrors {
+
+  private val validTypes = Seq(STANDARD, SUPPLEMENTARY, SIMPLIFIED, OCCASIONAL)
+
+  def displayPage(mode: Mode, itemId: String): Action[AnyContent] = (authenticate andThen journeyType(validTypes)) { implicit request =>
+    val formWithErrors = form.withSubmissionErrors
+
+    val frm = request.cacheModel.itemBy(itemId).flatMap(_.isLicenseRequired).fold(form.withSubmissionErrors) {
+      case true  => formWithErrors.fill(YesNoAnswer(yes))
+      case false => formWithErrors.fill(YesNoAnswer(no))
+    }
+
+    Ok(is_license_required(mode, itemId, frm, representativeStatusCode))
+
+  }
+
+  def submitForm(mode: Mode, itemId: String): Action[AnyContent] = (authenticate andThen journeyType).async { implicit request =>
+    form.bindFromRequest
+      .fold(
+        formWithErrors => Future.successful(BadRequest(is_license_required(mode, itemId, formWithErrors, representativeStatusCode))),
+        yesNo =>
+          updateCache(yesNo, itemId) map { _ =>
+            navigator.continueTo(mode, nextPage(yesNo, itemId))
+        }
+      )
+  }
+
+  private def nextPage(yesNoAnswer: YesNoAnswer, itemId: String)(implicit request: JourneyRequest[AnyContent]): Mode => Call =
+    yesNoAnswer.answer match {
+      case YesNoAnswers.yes => AdditionalDocumentAddController.displayPage(_, itemId)
+      case YesNoAnswers.no if request.cacheModel.isAuthCodeRequiringAdditionalDocuments =>
+        AdditionalDocumentsController.displayPage(_, itemId)
```

Comment:
I think this should be `AdditionalDocumentAddController.displayPage(_, itemId)`?